### PR TITLE
DEV: Prepare plugin for deprecation.

### PR DESCRIPTION
**Pull request by romanrizzi**
We'll merge this plugin into core soon and need to make a couple of changes to avoid conflict. Changes are:

- Wrap the sitemap model in a module so core can define its own.
- Don't update sitemaps or add the path to the robots.txt file if the `publish_sitemap` setting is present, which is what's core is going to use.
- Use append instead of prepend for mounting the engine, so the same routes defined by core can have a higher priority, and we never hit this plugin's controller.
- Core will use the same table if present, so we add `if_not_exists: true` in the migration just in case conflicts can originate based on the call order.

### CODE REVIEW

**Review by pmusaraj — COMMENTED** [review]
Looks generally good to me. 

Would it be useful to add a check to `plugin.rb` that skips everything if core already has sitemap support (i.e. if the `publish_sitemaps` setting is defined)? 

Also, minor, but would be good to update the readme too, to note that this plugin is going to be folded in core.

**Comment by davidtaylorhq** [comment]
> Would it be useful to add a check to plugin.rb that skips everything if core already has sitemap support (i.e. if the publish_sitemaps setting is defined)?

That sounds like a good idea to me. Plus, I think we should print a message to the admin dashboard, so that self-hosters are told to remove the plugin from their `app.yml`. Here's an example of where we've done that in the past: https://github.com/discourse/discourse-slack-official/blob/3d2a0210cde2e4dcb5dd7bb14e25536b7b6631f1/plugin.rb#L13-L16

**Comment by davidtaylorhq** [comment]
Now that it disables itself, could we skip all the renaming of modules/files? I imagine it no longer matters, since the plugin won't try to define them if core has the `publish_sitemaps` setting?

**Review by pmusaraj — APPROVED** [review]

File: plugin.rb [hunk 1]
```diff
@@ -11,83 +11,90 @@
...
+        render plain: @output, content_type: 'text/xml; charset=UTF-8' unless performed?
       end
-      render plain: @output, content_type: 'text/xml; charset=UTF-8' unless performed?
     end
-  end
 
-  Discourse::Application.routes.prepend do
-    mount ::DiscourseSitemap::Engine, at: "/"
-  end
+    Discourse::Application.routes.append do
```
Comment (davidtaylorhq) [inline plugin.rb]: Is this still required now that we auto-disable the plugin?

File: app/models/discourse_sitemap/sitemap.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
```
Comment (davidtaylorhq) [inline app/models/discourse_sitemap/sitemap.rb:1]: Is the rename of this class still required now that we auto-disable the plugin?

File: app/models/discourse_sitemap/sitemap.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+module DiscourseSitemap
+  class Sitemap < ::ActiveRecord::Base
```
Comment (davidtaylorhq) [inline app/models/discourse_sitemap/sitemap.rb]: Is the rename of this class still required now that we auto-disable the plugin?

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**